### PR TITLE
Support `true` and `false` as boolean flag params

### PR DESCRIPTION
**Pull request by Noratrieb**
Implements [MCP 577](https://github.com/rust-lang/compiler-team/issues/577).

### CODE REVIEW

File: compiler/rustc_session/src/options.rs [hunk 1]
```diff
@@ -432,11 +432,11 @@ mod parse {
     /// Use this for any boolean option that has a static default.
     pub(crate) fn parse_bool(slot: &mut bool, v: Option<&str>) -> bool {
```
Comment (compiler-errors) [inline compiler/rustc_session/src/options.rs:433]: should also implement this for `parse_opt_bool`

Comment (Nilstrieb) [reply compiler/rustc_session/src/options.rs:433]: excellent idea, ~~time to write another MCP for that as well~~

**Comment by ehuss** [comment]
Can you also update the documentation at https://github.com/rust-lang/rust/blob/master/src/doc/rustc/src/codegen-options/index.md?

**Review by wesleywiser — APPROVED** [review]
r=me with the docs updated

**Comment by wesleywiser** [comment]
The MCP seems uncontroversial in my opinion, but no reason not to

@rfcbot fcp merge

**Review by petrochenkov — APPROVED** [review]

**Review by wesleywiser — APPROVED** [review]

**Pull request merged**